bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -146,7 +146,10 @@ public Unit invoke(Boolean connected) {
                 sessionStore, logger);
         systemBroadcastReceiver = new SystemBroadcastReceiver(this, logger);
 
-        metadataState = configuration.metadataState.copy(configuration.metadataState.getMetadata());
+        Metadata orig = configuration.metadataState.getMetadata();
+        Metadata copy = orig.copy(orig.toMap(), orig.getJsonStreamer(), orig.getRedactedKeys());
+        metadataState = configuration.metadataState.copy(copy);
```

Comment:
This logic is confusing – it seems like the thing that needs to happen here is a copy of the metadata, but `copy` is mentioned 4x as well as `getMetaData()` and `toMap()`. Maybe it just needs to be this gnarly but if so can you add comments as to why and what it's doing?

For instance, why can you call `orig.copy()` but then it looks like you have to pass in the data that you want to copy… `orig.toMap()`?

I also don't see what this change has to do with the description of the PR?

Reply:
I've rebased and updated this to avoid copying like this.